js/classes/entities/midground/chad.js

```diff
@@ -1,58 +1,392 @@
+     * @param {GAME} running GAME.movingLeftShift, user intends for Chad to run.
+     * @param {GAME} dashingInEitherDirection GAME.running, user intends to dash mid-air.
+     * @param {Integer} toggleSign either a 1 or -1 to change the velocity direction.
+     * @returns xVelocity so that Chad's velocity can be updated.
+     * @author Caleb Krauter
+     */
+    manageXDirectionMovement(direction, xVelocity, walkInDirection, running, dashing) {
+        if (this.isOnGround) {
+            this.hasDoubleJumped = false;
+            this.canDash = true;
+            this.hasDashed = false;
+        }
+        const toggleSign = this.facing === "left" ? -1 : 1;
+
+        // Walk action
+        this.action = "walking";
+        this.facing = direction;
+        xVelocity += toggleSign * Chad.SPEED;
+
+        // Run action
+        if (running) {
+            this.action = "running";
+            this.facing = direction;
+            xVelocity += toggleSign * Chad.SPEED * Chad.SPRINT_MULTIPLIER;
+        }
+        // Dash action
+        // When not CHAD is not on the ground and his dash is allowed and the user is trying to dash then enter conditional.
+        if (dashing && walkInDirection && !this.isOnGround && this.canDash && !this.hasDashed) {
+            if (!this.isDashing) {
+                this.xDashAnchoredOrigin = this.pos.x;
+                this.isDashing = true
+            }
+            this.action = "dashing";
+            xVelocity += toggleSign * Chad.SPEED * Chad.DASH_MULTIPLIER;
+            // Used to limit the distance of the dash.
+            let deltaX = Math.abs(this.pos.x - this.xDashAnchoredOrigin);
+            // Limit the delta in x that CHAD can dash. Set booleans as necessary to ensure
+            // correct limitations on dash functionality, i.e. no double dashing, no infinite dash.
+            if (deltaX >= this.DASH_BARRIER) {
+                this.canDash = false;
+                this.hasDashed = true;
+                this.isDashing = false;
+            } else {
+                this.canDash = true;
+                this.hasDashed = false;
+                this.isDashing = true;
+            }
+        }
+        // Prevents continuing a dash after lifting the dash key.
+        if (!GAME.running && !this.isOnGround && this.isDashing) {
+            this.canDash = false;
+            this.hasDashed = true;
+            this.isDashing = false;
+        }
+        return xVelocity;
+    }
+
+    /**
+     * Deals with movement in the y direction including all types of jumping.
+     * 
+     * @param {Integer} yVelocity Chad's y velocity.
+     * @param {Integer} deltaHeight the change in distance from the origin of 
+     *                              Chad's second jump and his current height.
+     * @returns yVelocity so that Chad's velocity can be updated.
+     */
+    manageYDirectionMovement(yVelocity, deltaHeight) {
```

Comment:
neither of these parameters really need to get passed in either. Remember you can use the references you have inside the method body to calculate those really quickly!